src/tpm.rs

```diff
@@ -123,6 +128,129 @@ pub(crate) fn create_ek(
 assert_eq_size!(TPML_PCR_SELECTION, [u8; 132]);
 assert_eq_size!(TPML_DIGEST, [u8; 532]);
 
+// Serialize a TPML_PCR_SELECTION into a Vec<u8>
+// The serialization will adjust the data endianness as necessary and add paddings to keep the
+// memory aligment.
+pub(crate) fn serialize_pcrsel(
+    pcr_selection: &TPML_PCR_SELECTION,
+) -> Vec<u8> {
+    let mut output = Vec::with_capacity(TPML_PCR_SELECTION_SIZE);
+    output.extend(u32::to_le_bytes(pcr_selection.count));
+    for selection in pcr_selection.pcrSelections.iter() {
+        output.extend(selection.hash.to_le_bytes());
+        output.extend(selection.sizeofSelect.to_le_bytes());
+        output.extend(selection.pcrSelect);
+        output.extend([0u8; 1]); // padding to keep the memory aligment
+    }
+    output
+}
+
+// Deserialize a TPML_PCR_SELECTION from a &[u8] slice.
+// The deserialization will adjust the data endianness as necessary.
+pub(crate) fn deserialize_pcrsel(
+    pcrsel_vec: &[u8],
+) -> Result<TPML_PCR_SELECTION> {
+    if pcrsel_vec.len() != TPML_PCR_SELECTION_SIZE {
+        return Err(KeylimeError::InvalidRequest);
+    }
+
+    let mut reader = std::io::Cursor::new(pcrsel_vec);
+    let mut count_vec = [0u8; 4];
+
+    let len = reader.read(&mut count_vec)?;
+    if len < count_vec.len() {
+        return Err(KeylimeError::InvalidRequest);
+    }
```

Comment:
Would it make sense to use `read_exact` as it returns dedicated error on unexpected EOF? Something like:
```suggestion
    reader.read_exact(&mut count_vec)?;
```
for other places as well.

Reply:
Thanks for the suggestion, I'll replace the reads with `read_exact()`